### PR TITLE
fix: Fixes read & backend replicas settings

### PR DESCRIPTION
**Pull request by thorker**
**What this PR does / why we need it**:
Fixes incorrect configuration of read & backend replicas
**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: production/helm/loki/templates/backend/statefulset-backend.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ spec:
   {{- if eq .Values.deploymentMode "SingleBinary" }}
   replicas: 0
   {{- else }}
-  replicas: {{ .Values.write.replicas }}
+  replicas: {{ .Values.backend.replicas }}
```
Comment (MichelHollands) [inline production/helm/loki/templates/backend/statefulset-backend.yaml:26]: 4 lines above this still says `if not .Values.write.autoscaling.enabled`. Shouldn't that use backend as well?

Comment (thorker) [reply production/helm/loki/templates/backend/statefulset-backend.yaml:26]: Good point

File: production/helm/loki/Chart.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ name: loki
 description: Helm chart for Grafana Loki in simple, scalable mode
 type: application
 appVersion: 3.0.0
-version: 6.3.4
+version: 6.3.5
```
[the remark this answers is not in the document]
Comment (thorker) [reply production/helm/loki/Chart.yaml]: Thought README & CHANGELOG will be generated via CI Pipeline or did I missed sth?

[the remark this answers is not in the document]
Comment (MichelHollands) [reply production/helm/loki/Chart.yaml]: Not for the Helm charts at the moment so they have to be updated by hand.

**Review by MichelHollands — APPROVED** [review]

**Pull request merged**